src/extremeweatherbench/regions.py

```diff
@@ -1,55 +1,84 @@
 """Region classes and utilities for the ExtremeWeatherBench package."""
 
+import abc
 import dataclasses
 import logging
-from abc import ABC, abstractmethod
+import pathlib
 from collections import namedtuple
-from pathlib import Path
 from typing import Any, Type
 
 import geopandas as gpd  # type: ignore[import-untyped]
 import numpy as np
 import regionmask
+import shapely  # type: ignore[import-untyped]
 import xarray as xr
-from shapely import MultiPolygon, Polygon  # type: ignore[import-untyped]
 
 from extremeweatherbench import utils
 
 logger = logging.getLogger(__name__)
 
 
-class Region(ABC):
+class Region(abc.ABC):
     """Base class for different region representations."""
 
     @classmethod
-    @abstractmethod
+    @abc.abstractmethod
     def create_region(cls, *args, **kwargs) -> "Region":
         """Abstract factory method to create a region; subclasses must implement with
         their own, specialized arguments."""
         pass
 
     @property
-    @abstractmethod
+    @abc.abstractmethod
     def geopandas(self) -> gpd.GeoDataFrame:
         """Return representation of this Region as a GeoDataFrame."""
         pass
 
-    def mask(self, dataset: xr.Dataset, drop: bool = False) -> xr.Dataset:
-        """Mask a dataset to the region."""
-        mask = regionmask.mask_geopandas(
-            self.geopandas, dataset.longitude, dataset.latitude
-        )
-        mask_array = ~np.isnan(mask)
-        return dataset.where(mask_array, drop=drop)
-
     @property
     def get_bounding_coordinates(self) -> tuple[Any, ...]:
-        """Get the bounding coordinates of the region."""
+        """Get the bounding coordinates of the region.
+
+        Returns:
+            A named tuple with longitude_min, latitude_min, longitude_max, and
+            latitude_max.
+        """
         return namedtuple(
             "BoundingCoordinates",
             ["longitude_min", "latitude_min", "longitude_max", "latitude_max"],
         )(*self.geopandas.total_bounds)
```

Comment:
The typing specification is wrong here. If you want to keep things very simple, then the return type could simply be `tuple[int, int, int, int]`, and you would just pass the ints back directly. If you do want to use a struct, then you should do two things:

1. Move the `BoundingCoordinates` definition (the `namedtuple` call) to the module-level
2. Change the return type to `BoundingCoordinates`.